datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/ZookeeperBackedDatastreamStore.java

```diff
@@ -259,6 +260,61 @@ public void forceCleanupDatastream(String key) {
     }
   }
 
+  /**
+   * Creates or Updates the datastream entry in the persistent store with the latest list of violating topics.
+   * @param key Name of the datastream whose topics have to be handled.
+   * @param throughputViolatingTopics topics, of which at least one partition violates the brooklin's permissible
+   *                                 throughput thresholds.
+   */
+  @Override
+  public void createOrUpdateThroughputViolatingDatastreamEntry(String key, Set<String> throughputViolatingTopics)
+      throws DatastreamException {
+    String throughputViolatingDatastreamPath = KeyBuilder.throughputViolationsPerDatastream(_cluster, key);
+
+    // If a set of violating topics exists already, remove and create a new entry to trigger a child change notification.
```

Comment:
Will it make sense to have a config knob to control this feature at Brooklin level, rather than relying on external services to stop sending?